client/rest-high-level/src/main/java/org/elasticsearch/client/snapshotlifecycle/SnapshotLifecyclePolicy.java

```diff
@@ -61,11 +59,11 @@
         PARSER.declareString(ConstructingObjectParser.constructorArg(), NAME);
         PARSER.declareString(ConstructingObjectParser.constructorArg(), SCHEDULE);
         PARSER.declareString(ConstructingObjectParser.constructorArg(), REPOSITORY);
-        PARSER.declareObject(ConstructingObjectParser.constructorArg(), (p, c) -> p.map(), CONFIG);
+        PARSER.declareObject(ConstructingObjectParser.optionalConstructorArg(), (p, c) -> p.map(), CONFIG);
     }
 
     public SnapshotLifecyclePolicy(final String id, final String name, final String schedule,
-                                   final String repository, Map<String, Object> configuration) {
+                                   final String repository, @Nullable Map<String, Object> configuration) {
         this.id = Objects.requireNonNull(id);
         this.name = name;
         this.schedule = schedule;
```

Comment:
nit: it is unclear if `name`, `schedule`, and `repository` are `@Nullable` If so, can you add the annontation... if not can you Objects.requireNonNull()` ?